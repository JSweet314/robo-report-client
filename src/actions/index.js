export const submitNewUser = user => ({
  type: 'SUBMIT_NEW_USER',
  user
});

export const getSavedUserInfo = userEmail => ({
  type: 'GET_USER_INFO',
  userEmail
});

export const captureUser = user => ({
  type: 'CAPTURE_USER',
  user
});

export const submitNewComplaint = complaint => ({
  type: 'SUBMIT_NEW_COMPLAINT',
  complaint
});

export const captureComplaint = complaint => ({
  type: 'CAPTURE_COMPLAINT',
  complaint
});

export const getUserComplaints = userId => ({
  type: 'GET_USER_COMPLAINTS',
  userId
});

export const captureDbComplaints = complaints => ({
  type: 'CAPTURE_DB_COMPLAINTS',
  complaints
});

export const captureError = error => ({
  type: 'CAPTURE_ERROR',
  error
});

export const getFCCData = () => ({
  type: 'GET_FCC_DATA'
});

export const captureFCCData = fccData => ({
  type: 'CAPTURE_FCC_DATA',
  fccData
});