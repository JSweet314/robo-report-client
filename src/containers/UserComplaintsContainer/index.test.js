import React from 'react';
import { shallow } from 'enzyme';
import { 
  UserComplaintsContainer, 
  mapStateToProps, 
  mapDispatchToProps 
} from './index';
import * as actions from '../../actions';

describe('UserComplaintsContainer', () => {
  let wrapper;
  const mockComplaints = [{ subject: 'Nuisance Caller', id: 1 }];
  const mockFilterReports = jest.fn();
  const mockHistory = {};
  beforeEach(() => {
    wrapper = shallow(
      <UserComplaintsContainer 
        complaints={mockComplaints} 
        reportFilter={'ALL'}
        filterReports={mockFilterReports}
        history={mockHistory}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot if no complaints', () => {
    wrapper = shallow(
      <UserComplaintsContainer 
        complaints={[]} 
        reportFilter={'ALL'}
        filterReports={mockFilterReports}
        history={mockHistory}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    const mockState = { complaints: mockComplaints, reportFilter: 'ALL'};
    const mapped = mapStateToProps(mockState);

    it('should return complaints from the state', () => {
      expect(mapped)
        .toHaveProperty('complaints', mockComplaints);
    });

    it('should return reportFilter from the state', () => {
      expect(mapped)
        .toHaveProperty('reportFilter', 'ALL');
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    
    it('should map filterReports action to props', () => {
      mapped.filterReports('ALL');
      expect(mockDispatch).toHaveBeenCalledWith(actions.filterReports('ALL'));
    });
  });
});
