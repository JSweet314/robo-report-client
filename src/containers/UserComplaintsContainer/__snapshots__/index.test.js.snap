// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComplaintsContainer should match the snapshot 1`] = `
<div
  className="complaintsContainer"
>
  <h3
    className="complaintsContainer__heading"
  >
    My Reports
  </h3>
  <div
    className="complaintsContainer__group"
  >
    <SummaryReport
      key="complaint-1"
      questionBlocks={
        Array [
          Object {
            "headline": "Tell us about the call.",
            "questions": Array [
              Object {
                "label": "Subject",
                "required": true,
                "type": "text",
                "value": "subject",
              },
              Object {
                "label": "Description",
                "required": true,
                "type": "textarea",
                "value": "description",
              },
            ],
          },
          Object {
            "headline": "Business or pleasure?",
            "questions": Array [
              Object {
                "label": "Did the call/message that you are reporting advertise any type of property, goods, or services?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                ],
                "required": true,
                "type": "select",
                "value": "isSoliciting",
              },
              Object {
                "dependent": "isSoliciting",
                "label": "Type of Property, Goods, or Services",
                "required": true,
                "type": "text",
                "value": "typeOfSolicit",
              },
              Object {
                "dependent": "isSoliciting",
                "label": "Have you or anyone else in your household done any business with the caller / company within the past 18 months immediately before you received the call / message?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                  "Uncertain",
                ],
                "required": true,
                "type": "select",
                "value": "doneBusinessWith",
              },
              Object {
                "dependent": "isSoliciting",
                "label": "Have you or anyone else in your household made any inquiry or application to the caller/company within the 3 months immediately before you received the call/message?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                  "Uncertain",
                ],
                "required": true,
                "type": "select",
                "value": "inquiredWith",
              },
              Object {
                "dependent": "isSoliciting",
                "label": "Do you or anyone in your household have a personal relationship with the individual that made the call?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                  "Uncertain",
                ],
                "required": true,
                "type": "select",
                "value": "householdRelation",
              },
              Object {
                "label": "Have you or anyone else in your household or business given the caller/company permission to call?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                  "Uncertain",
                ],
                "required": true,
                "type": "select",
                "value": "permissionToCall",
              },
              Object {
                "dependent": "permissionToCall",
                "label": "Did you give permission to the caller/business to call you in writing?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                ],
                "required": false,
                "type": "select",
                "value": "writtenPermission",
              },
              Object {
                "dependent": "permissionToCall",
                "label": "Date you provided permission to the caller/company to call",
                "required": false,
                "type": "date",
                "value": "dateOfPermission",
              },
            ],
          },
          Object {
            "headline": "The nitty gritty",
            "questions": Array [
              Object {
                "label": "Date of Your Issue/Problem",
                "required": true,
                "type": "date",
                "value": "date",
              },
              Object {
                "label": "Time of Your Issue/Problem",
                "required": true,
                "type": "time",
                "value": "time",
              },
              Object {
                "label": "Type of Call or Message",
                "options": Array [
                  "-",
                  "Abandoned Calls",
                  "Live Voice",
                  "Prerecorded Voice",
                  "Text Message",
                ],
                "required": true,
                "type": "select",
                "value": "typeOfCall",
              },
              Object {
                "label": "Did you receive Caller ID Information?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                  "Don't have caller id",
                  "Uncertain",
                ],
                "required": true,
                "type": "select",
                "value": "receivedCallerId",
              },
              Object {
                "dependent": "receivedCallerId",
                "label": "Caller ID Number",
                "required": false,
                "type": "tel",
                "value": "callerIdNumber",
              },
              Object {
                "dependent": "receivedCallerId",
                "label": "Caller ID Name",
                "required": false,
                "type": "text",
                "value": "callerIdName",
              },
            ],
          },
          Object {
            "headline": "Conversational pleasantries",
            "questions": Array [
              Object {
                "label": "Was the caller's business name provided during the call/message?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                  "Uncertain",
                ],
                "required": true,
                "type": "select",
                "value": "receivedBusinessName",
              },
              Object {
                "dependent": "receivedBusinessName",
                "label": "Was the business name provided at the beginning of the call/message?",
                "options": Array [
                  "-",
                  "Yes",
                  "No",
                ],
                "required": true,
                "type": "select",
                "value": "nameAtBeginning",
              },
              Object {
                "dependent": "receivedBusinessName",
                "label": "Please provide the name of the advertiser provided during the call",
                "required": true,
                "type": "text",
                "value": "providedAdvertiserName",
              },
              Object {
                "label": "Provide the advertiser's phone number given during the call.",
                "required": false,
                "type": "tel",
                "value": "providedAdvertiserNumber",
              },
              Object {
                "label": "Addtl. Info",
                "required": false,
                "type": "textarea",
                "value": "addtlInfo",
              },
            ],
          },
        ]
      }
      values={
        Object {
          "id": 1,
          "subject": "Nuisance Caller",
        }
      }
    />
  </div>
</div>
`;
